provisioner/file/provisioner.go

```diff
@@ -154,7 +159,11 @@ func (p *Provisioner) ProvisionUpload(ui packer.Ui, comm packer.Communicator) er
 			return err
 		}
 
-		err = comm.Upload(p.config.Destination, f, &fi)
+		if strings.HasSuffix(dst, "/") {
+			dst = filepath.Join(dst, filepath.Base(src))
```

Comment:
Can you tell me if this changes the behavior at all? It looks like it works like before to me, but wanted to understand the reason for the change